opus/application/static_media/js/browse.js

```diff
@@ -796,6 +732,187 @@ var o_browse = {
+
+        // table: obsNum = alignedCachedFirstObs + number of row
+        // gallery: obsNum = alignedCachedFirstObs + number of row * number of obs in a row
+        // Note: in table view, if there are more than one row, we divide by the 2nd table tr's
+        // height because in some corner cases, the first table tr's height will be 1px larger
+        // than rest of tr, and this will mess up the calculation.
+        let tableRowHeight = ($(`${tab} tbody tr`).length === 1 ? $(`${tab} tbody tr`).outerHeight() :
+                              $(`${tab} tbody tr`).eq(1).outerHeight());
+        let obsNumDiff = (o_browse.isGalleryView() ?
+                          Math.round((topBoxBoundary - firstCachedObsTop)/o_browse.imageSize) *
+                          galleryBoundingRect.x :
+                          Math.round((topBoxBoundary - firstCachedObsTop)/
+                          tableRowHeight));
+
+        let obsNum = obsNumDiff + alignedCachedFirstObs;
+
+        // currentScrollObsNum: the current top item in table view. This item is
+        // also one of the top row items in gallery view
+        // (it will be used to updated the table view scrollbar location).
+        let currentScrollObsNum = obsNum;
+
+        // obsNum: startObs, the most top left obsNum in gallery.
+        // (it will be used to updated slider obsNum).
+        // The calculation below is to make sure we are getting the first item in the
+        // top row in gallery view.
+        obsNum = (o_utils.floor((obsNum - 1)/galleryBoundingRect.x) *
+                  galleryBoundingRect.x + 1);
+
+        // In gallery view, if scrollbarObsNum in infiniteScroll instance is:
+        // (1) still within the current startObs' boundary
+        // (2) larger than max slider value when current startObs is equal to max slider value
+        // We want to make sure it didn't get updated to startObs. This will make sure table
+        // view scrollbar location stays at where it's been left off when we switch back to
+        // table view again.
+        let nextObsNum = obsNum + galleryBoundingRect.x;
+        let contentsView = o_browse.getScrollContainerClass();
+        let previousScrollObsNum = $(`${tab} ${contentsView}`).data("infiniteScroll").options.scrollbarObsNum;
+        let maxSliderVal = o_browse.getSliderMaxValue();
+        obsNum = Math.min(obsNum, maxSliderVal);
+        if ((previousScrollObsNum > obsNum && previousScrollObsNum < nextObsNum) ||
+            (obsNum === maxSliderVal && previousScrollObsNum > maxSliderVal)) {
+            currentScrollObsNum = o_browse.isGalleryView() ? previousScrollObsNum : currentScrollObsNum;
+        }
+
+        return {"startObs": obsNum, "scrollbarObsNum": currentScrollObsNum};
+    },
+
+    deleteObsToCorrectRowBoundary: function(tab, firstCachedObs) {
+        /**
+         * If browser window is resized, delete some cached observations
+         * to make sure row boundary is correct.
```

Comment:
This is such a weird thing to do that I think it would be helpful to have a much more descriptive comment so future developers will understand why we're doing this. Say something about the automatic wrapping around of the thumbnails and how when the browser is resized the first element in the cache won't necessarily be on an appropriate row boundary anymore, and we have to delete obs to make that true, because we always want the cache to be on a row boundary.